File: examples/plugins/temperature.js
// tio-gui analysis plugin API 1: one retained log entry, no device side effects.
function transform(input) {
  const match = /temp(?:erature)?\s*[:=]\s*(-?\d+(?:\.\d+)?)/i.exec(input);
  if (!match) return {error: 'No temperature field in this entry'};
  const celsius = Number(match[1]);
  return {celsius, fahrenheit: celsius * 9 / 5 + 32};
}

File: examples/plugins/temperature.lua
-- tio-gui analysis plugin API 1. Return UTF-8 text.
function transform(input)
  local temperature = tonumber(input:match('temp%s*[:=]%s*([%-%.%d]+)'))
  if not temperature then return 'No temp field in this entry' end
  return string.format('Celsius: %.2f; Fahrenheit: %.2f', temperature, temperature * 9 / 5 + 32)
end
